File: python/01/main01.py
"""
Advent of Code 2023
2023 December 01

Frank Schoeneman

"""

def extract_cal_code(line):

    only_digits = [this_char for this_char in line \
                                    if this_char.isdigit()]

    first_digit = only_digits[0]
    last_digit  = only_digits[-1]

    return int(first_digit + last_digit)


if __name__ == "__main__":

    input_file = "input.txt"

    with open(input_file, 'r') as f:
        cal_lines = f.readlines()

    print( sum( map(extract_cal_code, cal_lines) ) )

File: rust/01/main01.rs
use std::fs::File;
use std::io::{self, BufRead};

// Advent of Code 2023
// 2023 December 01
// Frank Schoeneman


fn find_cal_codes(file_path: &str) -> io::Result<i32> {

    let file = File::open(file_path)?;
    let reader = io::BufReader::new(file);

    let mut total = 0;

    for line in reader.lines() {

        let line = line?;
        let digits: Vec<char> = line.chars().filter(|c| c.is_digit(10)).collect();

        let first_digit = digits[0].to_string();
        let last_digit = digits.last().unwrap().to_string();
 
        let concat_digits = format!("{}{}", first_digit, last_digit);
        let line_code: i32 = concat_digits.parse().unwrap();

        total += line_code;

    }
    Ok(total)

}

fn main() {

    let file_path = "input.txt";
    match find_cal_codes(file_path){
        Ok(total) => println!("Total: {}", total), Err(err) => eprintln!("Error: {}", err),
    }
}
